packages/std-client/src/setup/setupMUDNetwork.ts

```diff
@@ -156,3 +165,18 @@ export async function setupMUDNetwork<C extends ContractComponents, SystemTypes
     components,
   };
 }
+
+/**
+ * Find a component in the components object by contract id, or return the component if it doesn't exist
+ * @param components object of components
+ * @param component component to find
+ * @returns component if it exists in components object, otherwise the component passed in
+ */
+function findOrDefineComponent<Cs extends ContractComponents, C extends ContractComponent>(
+  components: Cs,
+  component: C
+): C {
+  return (
+    (Object.values(components).find((c) => c.metadata.contractId === component.metadata.contractId) as C) ?? component
+  );
+}
```

Comment:
this approach is needed to enforce the component types (ie `SystemRegistry` needs to have the `SystemRegistry` type, a general `C[keyof C]` is not strict enough)